R/initialisers.R

```diff
@@ -0,0 +1,108 @@
+      Please enter a 10 digit code as a character string.
+      e.g. "0123QWERTY"'
+    )
+  }
+
+  html_script <- "<script>
+// Define dataLayer and the gtag function.
+window.dataLayer = window.dataLayer || [];
+function gtag(){dataLayer.push(arguments);}
+
+// Default ad_storage to 'denied' as a placeholder
+// Determine actual values based on your own requirements
+gtag('consent', 'default', {
+  'ad_storage': 'denied',
+  'analytics_storage': 'denied'
+});
+</script>
+
+<!-- Global site tag (gtag.js) - Google Analytics -->
+  <script async src='https://www.googletagmanager.com/gtag/js?id=G-XXXXXXXXXX'></script>
+  <script>
+    window.dataLayer = window.dataLayer || [];
+    function gtag(){dataLayer.push(arguments);}
+
+    gtag('js', new Date());
+
+  gtag('config', 'G-XXXXXXXXXX');
+
+/*
+The custom trackers below can be tailored to match the inputs used in your dashboard.
+*/
+
+  $(document).on('change', 'select#selectPhase', function(e) {
+    gtag('event', 'select phase', {'event_category' : 'choose Phase',
+    'event_label' : document.querySelector('select#selectPhase').value
+    });
+  });
+
+  $(document).on('change', 'select#selectArea', function(e) {
+    gtag('event', 'select area', {'event_category' : 'choose Area',
+    'event_label' : document.querySelector('select#selectArea').value
+    });
+  });
+
+  $(document).on('click', 'ul#navlistPanel', function(e) {
+    gtag('event', 'navlistPanel', {'event_category' : 'navbar click',
+    'event_label' : document.querySelector('ul#navlistPanel > li.active > a').getAttribute('data-value')
+    });
+  });
+
+  $(document).on('click', 'ul#tabsetpanels', function(e) {
+    gtag('event', 'tab panels', {'event_category' : 'tab panel clicks',
+    'event_label' : document.querySelector('ul#tabsetpanels > li.active > a').getAttribute('data-value')
+    });
+  });
+
+
+  $(document).on('click', 'a#download_data', function(e) {
+    gtag('event', 'Download button', {'event_category' : 'Download button click'
+    });
+  });
+
+  $(document).on('shiny:disconnected', function(e) {
+    gtag('event', 'disconnect', {
+      'event_label' : 'Disconnect',
+      'event_category' : 'Disconnect'
+    });
+  });
+
+</script>
+      " %>% gsub("XXXXXXXXXX", ga_code, .)
+  if (file.exists("google-analytics.html")) {
+    message("Analytics file already exists. If you have any customisations in that file, make sure you've backed those up before over-writing.")
+    user_input <- stringr::str_trim(readline(prompt = "Are you happy to overwrite the existing analytics script (y/N) "))
+    if (user_input %in% c("y", "Y")) {
+      write_out <- TRUE
+    } else {
+      write_out <- FALSE
+    }
+  } else {
+    write_out <- TRUE
+  }
+  if (write_out) {
+    cat(html_script, file = "google-analytics.html")
+    message("Google analytics script created in google-analytics.html.")
+    message("You'll need to add the following line to your ui.R script to start
+            recording analytics:")
```

Comment:
Could we leave this as a long line and put `# nolint: [line_length_linter]` or use paste0() to split the lines without the extra indent it's giving in the console currently?

![image](https://github.com/dfe-analytical-services/dfeshiny/assets/52536248/9f17ee26-e144-452a-ab89-a85c55686bfb)

Reply:
Changed my mind as this will take it over the 100 character limit for CRAN